src/main/java/net/mcreator/generator/LocalizationUtils.java

```diff
@@ -127,7 +127,20 @@ public static void generateLocalizationKeys(Generator generator, GeneratableElem
 		return keysToEntries;
 	}
 
-	private static void addLocalizationEntry(Generator generator, String key, Map<?, ?> template, Object entry) {
+	public static boolean shouldBeSkippedBasedOnCondition(Workspace workspace, Generator generator,
+			Map<?, ?> template) {
+		TemplateExpressionParser.Operator operator = TemplateExpressionParser.Operator.AND;
+		String conditionRaw = (String) template.get("condition");
+		if (conditionRaw == null) {
+			conditionRaw = (String) template.get("condition_any");
+			operator = TemplateExpressionParser.Operator.OR;
+		}
+		return TemplateExpressionParser.shouldSkipTemplateBasedOnCondition(generator, conditionRaw,
+				workspace.getWorkspaceInfo(), operator);

```

Comment:
You are basing your conditions on data held by mod elements but pass the workspace overview here instead of the appropriate entry, not sure how is this even supposed to work.

Reply:
Not sure what to put as `conditionData` then

Reply:
Pass an `entry` `Object` the condition is defined for.